### PR TITLE
[RTD-615] Added delete app logic + unit tests

### PR DESCRIPTION
**Pull request by petretiandrea**
#### Description

This PR aims to adds business logic for a delete endpoint.

#### List of Changes

- added delete interface to remove existing sender code association's
- unit test
- narrow integration between mongodb and senderauth service business logic (no rest)

#### How Has This Been Tested?

- [x] Unit Test Suite
- [x] Integration Test Suite
- [ ] Api Tests
- [ ] Load Tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by TommasoLencioni — COMMENTED** [review]
Good job, thank you!

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/service/SenderAuthServiceImpl.java [hunk 1]
```diff
@@ -42,4 +43,18 @@ public void saveApiKey(String senderCode, String apiKey) {
             }
         }
     }
+
+  @Override
+  public void deleteAssociation(String senderCode, String apiKey) {
+    final var apiKeyAssociations = senderAuthRepository.findByApiKey(apiKey).orElse(null);
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/service/SenderAuthServiceImpl.java:49]: Imho you can the Optional library. It's the typical use case, the repo returns a Optional and in the next `if` you check with `optional.isPresent()`.

Comment (petretiandrea) [reply src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/service/SenderAuthServiceImpl.java:49]: I generally prefer to fold any functional object like Optional. It's a challenge between `Objects.nonNull(apiKeyAssociations)` and `apiKeyAssociations.isPresent()`. Nevertheless, I hate handling nullable objects, so I'm agree with you `isPresent` fits better. Thanks

Comment (and-mora) [reply src/main/java/it/gov/pagopa/rtd/ms/rtdmssenderauth/service/SenderAuthServiceImpl.java:49]: Exactly for this reason i usually prefer `Optional`, null checks should be avoided when possible. Thank you

**Review by and-mora — APPROVED** [review]

**Review by and-mora — APPROVED** [review]
Good job!

**Pull request merged**